appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/AppsTransferPresenter.kt

```diff
@@ -37,5 +61,24 @@ class AppsTransferPresenter : BasePresenter<IAppsTransferView>(), IAppsTransferP
         view?.close()
     }
 
+    private fun parsePublicAddressData(resultCode: Int, intent: Intent?) {
+        transferManager.processResponse(resultCode, intent, object : TransferManager.AccountInfoResponseListener {
+            override fun onCancel() {
+                Log.d("AppInfoPresenter", "Operation cancelled, no public address received")
+            }
 
+            override fun onError(error: String) {
+                //TODO error cant parse address
+                view?.onTransferError()
+            }
+
+            override fun onResult(address: String) {
+                repository.storeReceiverAppPublicAddress(address)
```

Comment:
This is not a good way of passing back the result of an operation.
The method `processResponse` should either return the public address or take as a parameter a callback that returns the public address.

Reply:
done